File: backend/Charlotte/Charlotte/Model/Result.cs
namespace Charlotte.Model
{
    public class Result<T>
    {
        public int ResultCode { get; set; }
        public string ResultMessage { get; set; }
        public T? Data { get; set; }
        public Result(int resultCode, string resultMessage) 
        {
            this.ResultCode = resultCode;
            this.ResultMessage = resultMessage;
        }
    }
}

File: app/app/src/main/java/com/example/charlotte/model/Result.kt
package com.example.charlotte.retrofitService
// 只有訊息
data class ResultMessage(
    val message: String,
    val code: Int,
)
// 訊息跟資料
data class ResultData<T>(
    val message: String,
    val code: Int,
    val data: T
)
